atomic_reactor/plugins/exit_sendmail.py

```diff
@@ -185,14 +220,21 @@ def _render_mail(self, rebuild, success, auto_canceled, manual_canceled):
             'user': '<autorebuild>' if rebuild else self.submitter,
             'logs': url
         }
-        return (subject_template % formatting_dict, body_template % formatting_dict)
 
-    def _send_mail(self, receivers_list, subject, body):
+        full_logs = []
+        if rebuild and endstate == 'Failed':
+            full_logs = self._fetch_full_logs()
+
+        return (subject_template % formatting_dict, body_template % formatting_dict, full_logs)
+
+    def _send_mail(self, receivers_list, subject, body, full_logs=[]):
```

Comment:
Generally it's preferable to have None as the default value for parameters instead of `[]`, since lists are mutable. When using it later you can use `for entry in full_logs or []:`.

Reply:
Fixed.